packages/itwin/measure-tools/src/api/Measurement.ts

```diff
@@ -285,6 +329,67 @@ export abstract class Measurement {
     this.onTransientIdChanged(prevId);
   }
 
+  public get drawingMetaData(): DrawingMetaData | undefined {
+    return this._drawingMetaData;
+  }
+
+  public set drawingMetaData(data: DrawingMetaData | undefined) {
+    this._drawingMetaData = data;
+    this.onDrawingMetaDataChanged();
+  }
+
+  public set sheetViewId(id: string | undefined) {
+    this._sheetViewId = id;
+  }
+
+  public get sheetViewId(): string | undefined {
+    return this._sheetViewId;
+  }
+
+  public set drawingId(id: string | undefined) {
+    if (this._drawingMetaData) {
```

Comment:
Having setters for individual properties seem like a slippery slope if they don't end up setting anything.
I'd be in favor of removing all the individual getters/setters except for the `get/set` for the whole drawing metadata object.
That still leaves the possibility of misusing the API by assigning directly inside the object, which would bypass the `onDrawingMetadataChanged` event. (eg. `this.drawingMetaData.sheetViewId = "0x1234").
A comment on the getter is probably sufficient for this, but you can also return Readonly<DrawingMetaData> to enforce it.

To make it easier to change individual properties on the object, you can implement a 'with' function in the DrawingMetaData namespace, eg:

```ts
// Returns a new DrawingMetaData object with one or more properties changed.
export function with(current: DrawingMetaData, overrides?: Partial<DrawingMetaData>): DrawingMetaData {
  return { ...current, ...overrides };
} 
```

Reply:
I removed the getters/setters (except for the worldScale getter) and added readonly to the drawingMetadata getter.

I'm not sure about the "with" function and would like to know more about it if you have time.